### ISSUE TITLE
[docs] Add an example illustrating how to set schema validators per validation type

### ISSUE THREAD

**Issue by jsumners** [fastify/fastify#3262]
To anyone that finds this from a google search, here's my solution as of v3.19.1

```typescript
const schemaCompilers: Record<string, Ajv.Ajv> = {
    'body': new Ajv({
        removeAdditional: false,
        coerceTypes: false,
        allErrors: true
    }),
    'params': new Ajv({
        removeAdditional: false,
        coerceTypes: true,
        allErrors: true
    }),
    'querystring': new Ajv({
        removeAdditional: false,
        coerceTypes: true,
        allErrors: true,
    })
};

server.setValidatorCompiler(req => {
    if (!req.httpPart) { throw new Error('Missing httpPart'); }
    const compiler = schemaCompilers[req.httpPart];
    if (!compiler) { throw new Error(`Missing compiler for ${req.httpPart}`); }

    return compiler.compile(req.schema);
});
```

_Originally posted by @KrakerXyz in https://github.com/fastify/help/issues/128#issuecomment-897617445_

**Comment by jsumners** [comment]
Yeah, a test or two wouldn't be a bad idea.

**Comment by climba03003** [comment]
Resolved by #3310
